File: backend/app/schemas/application.py
import uuid

from pydantic import BaseModel, Field

from app.models.application import ApplicationStatus


class ApplicationCreate(BaseModel):
    care_request_id: uuid.UUID
    nurse_id: uuid.UUID
    message: str | None = Field(default=None, max_length=1000)


class ApplicationRejectRequest(BaseModel):
    reason: str | None = Field(default=None, max_length=500)


class ApplicationResponse(BaseModel):
    id: uuid.UUID
    care_request_id: uuid.UUID
    nurse_id: uuid.UUID
    patient_id: uuid.UUID
    status: ApplicationStatus
    message: str | None
    rejection_reason: str | None

    model_config = {"from_attributes": True}

File: mobile/lib/models/application.dart
import 'enums.dart';

class ApplicationInfo {
  final String id;
  final String careRequestId;
  final String nurseId;
  final String patientId;
  final ApplicationStatus status;
  final String? message;
  final String? rejectionReason;

  const ApplicationInfo({
    required this.id,
    required this.careRequestId,
    required this.nurseId,
    required this.patientId,
    required this.status,
    this.message,
    this.rejectionReason,
  });

  factory ApplicationInfo.fromJson(Map<String, dynamic> j) => ApplicationInfo(
        id: j['id'] as String,
        careRequestId: j['care_request_id'] as String,
        nurseId: j['nurse_id'] as String,
        patientId: j['patient_id'] as String,
        status: applicationStatusFromApi(j['status'] as String),
        message: j['message'] as String?,
        rejectionReason: j['rejection_reason'] as String?,
      );
}
